yang/frr-zebra.yang

```diff
@@ -0,0 +1,1717 @@
+      description "Debug MPLS.";
+    }
+    leaf debug-vxlan {
+      type empty;
+      description "Debug VxLAN.";
+    }
+    leaf debug-pw {
+      type empty;
+      description "Debug pseudowires.";
+    }
+    leaf debug-dplane {
+      type empty;
+      description "Debug the dataplane subsystem.";
+    }
+    leaf debug-dplane-detail {
+      type empty;
+      description "Debug dataplane subsystem details.";
+    }
+    leaf debug-mlag {
+      type empty;
+      description "Debug MLAG.";
+    }
+  }
+
+  /*
+   * Main zebra container
+   */ 
+  container zebra {
+    description
+      "Data model for the Zebra daemon.";
+
+    leaf mcast-rpf-lookup {
+      type frr-zebra:mcast-rpf-lookup-mode;
+      default "mrib-then-urib";
+      description
+	"Multicast RPF lookup behavior.";
+    }
+
+    leaf ip-forwarding {
+      type boolean;
+      description
+	"IP forwarding status.";
+    }
+
+    leaf ipv6-forwarding {
+      type enumeration {
+	enum "unknown" {
+	  value -1;
+	  description
+	    "Unknown state.";
+	}
+	enum "off" {
+	  value 0;
+	  description
+	    "IPv6 forwarding disabled.";
+	}
+	enum "on" {
+	  value 1;
+	  description
+	    "IPv6 forwarding enabled.";
+	}
+      }
+      description
+	"IPv6 forwarding status.";
+    }
+
+    leaf workqueue-hold-timer {
+      type uint32 {
+        range "0..10000";
+      }
+      units "milliseconds";
+      default "10";
+      description
+        "Work-queue processing hold timer, in milliseconds.";
+    }
+
+    leaf zapi-packets {
+      type uint32 {
+        range "1..10000";
+      }
+      default "1000";
+      description
+        "Number of ZAPI packets to process before relinquishing
+        the main thread.";
+    }
+
+    container import-kernel-table {
+      description
+        "Parameters to use when importing IPv4 routes from a non-main kernel
+        routing table.";
+
+      leaf table-id {
+        type uint32 {
+          range "1..252";
+        }
+        description
+          "The kernel table id.";
+      }
+      leaf distance {
+        type uint32 {
+          range "1..255";
+        }
+        default "15";
+        description
+          "The admin distance to use for imported routes.";
+      }
+      leaf route-map {
+        type string;
+        description
+          "A route-map to filter imported routes.";
+      }
+    }
+
+    leaf allow-external-route-update {
+      type empty;
+      description
+        "Allow FRR-controlled routes to be overwritten by external processes";
+    }
+
+    leaf dplane-queue-limit {
+      type uint32 {
+        range "0..10000";
+      }
+      default "200";
+      description
+        "Limit on the number of updates queued to the dataplane subsystem.";
+    }
+
+    list vrf-vni-mapping {
+      description
+        "EVPN VNI mapping corresponding to a VRF.";
+      key "vrf-id";
+      leaf vrf-id {
+        type uint32;
+        description
+          "The identifier for a VRF.";
```

Comment:
VRF related types should be pulled out into an `frr-vrf.yang` module and typedef'd